Userland/Utilities/pls.cpp

```diff
@@ -32,43 +32,46 @@
 #include <LibCore/ArgsParser.h>
 #include <LibCore/File.h>
 #include <LibCore/GetPassword.h>
+#include <LibCore/DirIterator.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
 
+static constexpr mode_t EXPECTED_PERMS = (S_IFREG | S_IRUSR);
+
 // Function Definitions
 extern "C" int main(int arch, char** argv);
 int unveil_paths(const char*);
 
 // Unveil paths, given the current user's path and the command they want to execute
 int unveil_paths(const char* command)
 {
-    // Get the system path, split it and attempt to unveil all the paths.
-    // We do NOT error out on an invalid path
-    auto paths = String(getenv("PATH")).split(':');
     int num_unveils = 0;
-    char path_buf[256];
-
-    // Unveil each path
-    for (const auto& path : paths) {
-        if (unveil(path.characters(), "x") == 0)
-            num_unveils++;
-    }
 
-    // Now unveil the command
-    auto command_path = realpath(command, &path_buf[0]);
+    // Attempt to unveil the command first...
```

Comment:
Let's see how we go with this. Not entirely sure if this is considered "awkward" in how I'm doing it, though it should take into consideration most cases (i.e someone supplies a direct path e.g `pls ./quake` or a system program `pls whoami`)